### PR TITLE
adds extra contact information to user

### PR DESCRIPTION
**Pull request by markcsims**
Adding the following the GP2 users:

- [x] telephone number
- [x] telephone country code
- [x] secondary email

### CODE REVIEW

File: apps/gp2-server/src/controllers/user.controller.ts [hunk 1]
```diff
@@ -20,26 +27,31 @@ export default class Users implements UserController {
...
-    return this.fetchById(id);
+    return this.fetchById(id, currentUserId);
   }
 
   async fetch(options: gp2.FetchUsersOptions): Promise<gp2.ListUserResponse> {
     const { total, items: users } = await this.userDataProvider.fetch(options);
 
-    const items = total > 0 ? users.map(parseUserToResponse) : [];
+    const items =
+      total > 0 ? users.map((user) => parseUserToResponse(user)) : [];
```
Comment (jhonny-m) [inline apps/gp2-server/src/controllers/user.controller.ts:40]: that's the same as before no?

File: apps/gp2-server/src/routes/user.route.ts [hunk 2]
```diff
@@ -56,7 +56,9 @@ export const userRouteFactory = (userController: UserController): Router => {
   userRoutes.get<{ userId: string }>('/users/:userId', async (req, res) => {
     const { userId } = validateUserParameters(req.params);
 
-    const user = await userController.fetchById(userId);
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const loggedInUserId = req.loggedInUser!.id;
+    const user = await userController.fetchById(userId, loggedInUserId);
```
Comment (jhonny-m) [inline apps/gp2-server/src/routes/user.route.ts:61]: dont we know here if it it the same id? and we could just pass a boolean?

File: apps/gp2-server/src/controllers/user.controller.ts (hunk 1 above)
Comment (markcsims) [reply apps/gp2-server/src/controllers/user.controller.ts:40]: No. parseUserToResponse now takes 2 parameters the user, and the current user id.

Like this users.map(parseUserToResponse), parseUserToResponse will be called with the user as the first param, and the index as the second.

i.e. the parameters of the callback. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

File: apps/gp2-server/src/routes/user.route.ts (hunk 2 above)
Comment (markcsims) [reply apps/gp2-server/src/routes/user.route.ts:61]: Yes, I was looking to put all the logic in the controller and leave the routing to do routes. Doing it like this is consistent with the project and working groups controllers - where they too take the loggedInUserId

**Review by jhonny-m — APPROVED** [review]

**Pull request merged**